components/brave_sync/client/bookmark_change_processor_unittest.cc

```diff
@@ -1340,3 +1340,31 @@ TEST_F(BraveBookmarkChangeProcessorTest, IgnoreMetadataSet) {
   EXPECT_CALL(*sync_client(), SendSyncRecords("BOOKMARKS", _)).Times(0);
   change_processor()->SendUnsynced(base::TimeDelta::FromMinutes(10));
 }
+
+TEST_F(BraveBookmarkChangeProcessorTest, MigrateOrdersForPermanentNodes) {
+  EXPECT_EQ(change_processor()->GetPermanentNodeIndex(
+      model()->bookmark_bar_node()), 1);
+  EXPECT_EQ(change_processor()->GetPermanentNodeIndex(
+      model()->other_node()), 2);
+
+  using brave_sync::BookmarkChangeProcessor;
+  EXPECT_EQ(BookmarkChangeProcessor::FindMigrateSubOrderLength("1.0.1."), 5);
+  EXPECT_EQ(BookmarkChangeProcessor::FindMigrateSubOrderLength("1.12.12."), 7);
+  EXPECT_EQ(BookmarkChangeProcessor::FindMigrateSubOrderLength("1.0.10."), 6);
+
+  change_processor()->Start();
+
+  EXPECT_FALSE(change_processor()->HasPermanentNodeMigrated(
+      const_cast<bookmarks::BookmarkNode*>(model()->other_node())));
+  EXPECT_FALSE(change_processor()->HasPermanentNodeMigrated(
+      const_cast<bookmarks::BookmarkNode*>(model()->bookmark_bar_node())));
+
+  EXPECT_CALL(*sync_client(), SendSyncRecords("BOOKMARKS", _)).Times(0);
```

Comment:
Because initially I could not make the check how orders modification work, because beyond the orders are assigned on `resolved-sync-records` and for the test sync client is mocked. 
But I modified it with direct set orders meta to see its changes.
Now it is fixed.